Doc/library/sqlite3.rst

```diff
@@ -1425,33 +1440,39 @@ This section shows recipes for common adapters and converters.
 Controlling Transactions
 ------------------------
 
-The underlying ``sqlite3`` library operates in ``autocommit`` mode by default,
-but the Python :mod:`sqlite3` module by default does not.
-
-``autocommit`` mode means that statements that modify the database take effect
-immediately.  A ``BEGIN`` or ``SAVEPOINT`` statement disables ``autocommit``
-mode, and a ``COMMIT``, a ``ROLLBACK``, or a ``RELEASE`` that ends the
-outermost transaction, turns ``autocommit`` mode back on.
-
-The Python :mod:`sqlite3` module by default issues a ``BEGIN`` statement
-implicitly before a Data Modification Language (DML) statement (i.e.
-``INSERT``/``UPDATE``/``DELETE``/``REPLACE``).
-
-You can control which kind of ``BEGIN`` statements :mod:`sqlite3` implicitly
-executes via the *isolation_level* parameter to the :func:`connect`
-call, or via the :attr:`isolation_level` property of connections.
-If you specify no *isolation_level*, a plain ``BEGIN`` is used, which is
-equivalent to specifying ``DEFERRED``.  Other possible values are ``IMMEDIATE``
-and ``EXCLUSIVE``.
-
-You can disable the :mod:`sqlite3` module's implicit transaction management by
-setting :attr:`isolation_level` to ``None``.  This will leave the underlying
-``sqlite3`` library operating in ``autocommit`` mode.  You can then completely
-control the transaction state by explicitly issuing ``BEGIN``, ``ROLLBACK``,
-``SAVEPOINT``, and ``RELEASE`` statements in your code.
-
-Note that :meth:`~Cursor.executescript` disregards
-:attr:`isolation_level`; any transaction control must be added explicitly.
+The ``sqlite3`` module does not adhere to the transaction handling recommended
+by PEP 249.
+Instead of keeping a transaction open and requiring the user to use the
+:meth:`~Connection.commit` and :meth:`~Connection.rollback` methods,
+``sqlite3`` only implicitly opens new transactions before
+:meth:`~Cursor.execute` and :meth:`~Cursor.executemany` executes any of the
+following statements:
+
+* INSERT
+* UPDATE
+* DELETE
+* REPLACE
+
+In addition, any pending transaction is implicitly committed in
+:meth:`~Cursor.executescript`, before execution of the given SQL script.
+No other implicit transaction handling is performed.
+
+You can control which kind of ``BEGIN`` statements ``sqlite3`` implicitly
+executes via the :attr:`isolation_level` connection attribute.
+
+The ``sqlite3`` module lets the user choose bypass its transaction handling, by
+setting :attr:`isolation_level` to :const:`None`.
+This leaves the underlying SQLite library in autocommit mode,
+but also allows the user to perform any transaction handling using explicit SQL
+statements.
+The SQLite library autocommit mode can be queried using the
+:attr:`in_transaction` connection attribute.
+
+.. note::
+
+   PEP 249's autocommit concept must not be mistaken for SQLite's autocommit
+   mode.
+   Though related, they are different concepts with different semantics.
```

Comment:
If we say A, we must also say B? It feels like we give the users two loose ends here: there's a PEP 249 autocommit concept, and there's an SQLite autocommit concept; now, go figure it out yourselves :) Is that acceptable? I neither want to repeat PEP 249 stuff here, nor the SQLite docs. A possible enhancement is to add links to the relevant sections in the respcective docs.

Reply:
I added an explicit link to PEP 249; we could link the specific section using the syntax ``:pep:`PEP 249 <249#section-id>` ``.

IMO yes, we should also link the appropriate section in the SQLite docs here, rather than leaving the reader completely out to dry.

Reply:
Actually, I think we should defer adding this note until #93823 has landed. I'll remove it for now.